components/clm/src/main/histFileMod.F90

```diff
@@ -983,17 +989,19 @@ subroutine hist_update_hbuf(bounds)
     ! !LOCAL VARIABLES:
     integer :: t                   ! tape index
     integer :: f                   ! field index
+    integer :: numdims             ! number of dimensions
     integer :: num2d               ! size of second dimension (e.g. number of vertical levels)
     character(len=*),parameter :: subname = 'hist_update_hbuf'
     !-----------------------------------------------------------------------
 
     do t = 1,ntapes
 !$OMP PARALLEL DO PRIVATE (f, num2d)
        do f = 1,tape(t)%nflds
-          num2d = tape(t)%hlist(f)%field%num2d
-          if ( num2d == 1) then
+          numdims = tape(t)%hlist(f)%field%numdims
+          if ( numdims == 1) then   
              call hist_update_hbuf_field_1d (t, f, bounds)
           else
+             num2d = tape(t)%hlist(f)%field%num2d
              call hist_update_hbuf_field_2d (t, f, bounds, num2d)
           end if
        end do
```

Comment:
`numdims` is a local variable that can be overwritten by multiple threads (race hazard). I'll push an update:
```
-!$OMP PARALLEL DO PRIVATE (f, num2d)
+!$OMP PARALLEL DO PRIVATE (f, numdims, num2d)
        do f = 1,tape(t)%nflds
           numdims = tape(t)%hlist(f)%field%numdims
           if ( numdims == 1) then
```